application/src/main/java/net/corda/v5/application/persistence/PersistenceService.java

```diff
@@ -21,64 +21,42 @@ public interface PersistenceService {
      * There is an automatic transient failure retry mechanism backing this method which can sometimes lead to the persistence
      * service internally making multiple requests to persist the same entities. In order that the same entities are not
      * erroneously persisted more than once, a de-duplication mechanism is in place to filter duplicate requests that were
-     * already successful. In order for the persistence service to be able to employ this mechanism, one of the following
-     * criteria must be met:
-     * <ul>
-     *     <li>The JPA entity must have an id defined, binding a field to the primary key field in the table, and the value of
-     *         this field provided for the entity must be deterministic at the point the persistence operation is made.
-     *         For example the value should not be populated with a timestamp which would potentially change each time the method
-     *         is called. Note that fields other than the id field in the entity do not have to be deterministic in this case.
-     *         That is you could populate the entity with a stable id, and a different field could hold a timestamp. If this
-     *         persist operation required retries to complete, the value of the non-deterministic field would represent the
-     *         first successful persist operation, for example the first timestamp that made it to the database, even if
-     *         subsequent attempts to retry the persist operation follow.</li>
-     *     <li>For JPA entities with no defined id, the entity must be populated with deterministic values in all its fields.
-     *         If any field were to change if the method were to be called again, for example if it was populated with a
-     *         timestamp, the de-duplication mechanism would fail to recognise this was a duplicate request and more than
-     *         one entity may be persisted.</li>
-     * </ul>
-     * It is up to the caller of this function to ensure either one of the above criteria are fulfilled where de-duplication
+     * already successful. In order for the persistence service to be able to employ this mechanism, a unique, deterministic
+     * deduplication ID must be provided. This ID has a maximum size of 128 characters.
+     *
+     * It is up to the caller of this function to ensure that the above criteria is fulfilled where de-duplication
      * of retries is a hard requirement. Otherwise, the caller should expect duplication might occur.
      *
+     * @param deduplicationId A unique identifier used to deduplicate persistence requests. This identifier must be non-deterministic
```

Comment:
Maybe it's obvious enough in your docs, but I wonder would it be worth explicitly including an example using a deterministic, compile-time constant ID? My worry is someone will try using a random UUID

```
    /**
     * ...
     * For example:
     * <pre>
     * private static final MY_ENTITY_DEDUPE_ID = "entity1dedupeId"
     * 
     * private void persistEntity() {
     *     persistenceService.persist(MY_ENTITY_DEDUPE_ID, entity)
     * }
     * </pre>
     *
     */
```

Reply:
Added the following to make the API super-explicit:
```
* In order for the persistence service to be able to employ this mechanism, this ID must be:
 * <ul>
 *     <li>Unique within the context of the flow;</li>
 *     <li>Deterministic at the time of the function call (I.e., a UUID should not be used).</li>
 *     <li>128 characters or fewer in length.</li>
 * </ul>
 * <p>
 * For example:
 * <pre>
 * private static final MY_ENTITY_DEDUPE_ID = "entity1dedupeId"
 *
 * private void persistEntity() {
 *     persistenceService.persist(MY_ENTITY_DEDUPE_ID, entities)
 * }
 * </pre>
```